PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/utils/PluginUtil.java

```diff
@@ -485,38 +501,39 @@ public static boolean forceInstallPluginUsingMarketPlaceAsync(String pluginSymbo
 	 * @param marketplaceURL: Marketplace installation URL, for example, http://marketplace.eclipse.org/marketplace-client-intro?mpc_install=421 for Scala
 	 * @return
 	 */
-	public static boolean forceInstallPluginUsingMarketplace(String pluginSymbolicName, String marketplaceURL) {
+	public static boolean forceInstallPluginUsingMarketplace(String pluginSymbolicName, String marketplaceURL, String manualInstallURL) {
```

Comment:
`manualInstallURL` can be `class URL` here, which can force validation check.

Reply:
Agree in general. Tried and prefer not to change:
1. These APIs will not be exposed to users
2. A lot of URL strings are static final strings. Need to handle exception if using URL. It can work but the code will look pretty hack.

---
In reply to: [147417907](https://github.com/Microsoft/azure-tools-for-java/pull/1139#discussion_r147417907) [](ancestors = 147417907)